### ISSUE TITLE
loading a job with custom job type may return wrong job type

### ISSUE THREAD

**Issue by niklassiemer** [pyiron/pyiron_base#803]
As I discovered in https://github.com/pyiron/pyiron_gui/pull/158 loading of custom defined job type returns the pyiron internal job type if this had been `JobType.register`ed (i.e. is in the `JOB_CLASS_DICT`):
```python
from pyiron_base import Project, PythonTemplateJob
from pyiron_base._tests import ToyJob

class ToyJob(PythonTemplateJob):
    def __init__(self, project, job_name):
        """A toyjob which can be run() to test functionalities."""
        super(ToyJob, self).__init__(project, job_name)
        self.input['input_energy'] = 100

    def run_static(self):
        self.output["energy_tot"] = self.input["input_energy"]
        self.status.finished = True

pr = Project('some_tmp')
job = pr.create_job(ToyJob, 'name')
job.run()
print(job["TYPE"])
job=pr['name']
print(job['TYPE'])
>> The job name was saved and received the ID: 86
>> <class '__main__.ToyJob'>
>> <class 'pyiron_base._tests.ToyJob'>
```

The conversion happens in 
https://github.com/pyiron/pyiron_base/blob/69fd55229647bf36c4db78555b62d5635f6f3562/pyiron_base/storage/hdfio.py#L1300-L1320

For me this was absolutely unexpected! If I did a calculation using a specific class I want the _same_ class back if I load the job, regardless of which jobs are registered.

I suggest to rewrite the above function to
```python
   def import_class(self, class_name):
        """
        Import given class from fully qualified name and return class object.
        Args:
            class_name (str): fully qualified name of a pyiron class
        Returns:
            type: class object of the given name
        """
        internal_class_name = class_name.split(".")[-1][:-2]
        class_path = class_name.split()[-1].split(".")[:-1]
        class_path[0] = class_path[0][1:]
        module_path = ".".join(class_path)        
        try:
            return getattr(
                importlib.import_module(module_path),
                internal_class_name,
            )
        except ImportError:    
            if internal_class_name in self._project.job_type.job_class_dict:
                module_path = self._project.job_type.job_class_dict[internal_class_name]
                return getattr(
                    importlib.import_module(module_path),
                    internal_class_name,
                )
            else:
                 raise 

```

which changes the behavior to _trust_ the module path we have written to hdf before and fall back to use the class registered as `JobType` if this is not possible.

**Comment by niklassiemer** [comment]
Identified current behavior as intended in the last pyiron meeting. I would nonetheless add a logger info that we do not return the same class if such a thing happens (e.g. when using a custom 'Lammps' job from plain `pyiron_base` and then loading that job from atomistics...

**Comment by niklassiemer** [comment]
Closed by #817
